inc/saitam.h

```diff
@@ -2079,6 +2128,118 @@ typedef sai_status_t (*sai_set_tam_event_attribute_fn)(
         _In_ sai_object_id_t tam_event_id,
         _In_ const sai_attribute_t *attr);
 
+/**
+ * @brief Counter Subscription attributes
+ */
+typedef enum _sai_tam_counter_subscription_attr_t
+{
+    /**
+     * @brief Start of Attributes
+     */
+    SAI_TAM_COUNTER_SUBSCRIPTION_ATTR_START,
+
+    /**
+     * @brief TAM telemetry type object
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_TAM_TEL_TYPE
+     */
+    SAI_TAM_COUNTER_SUBSCRIPTION_ATTR_TEL_TYPE = SAI_TAM_COUNTER_SUBSCRIPTION_ATTR_START,
+
+    /**
+     * @brief Subscribed object
+     *
+     * @type sai_object_id_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     * @objects SAI_OBJECT_TYPE_BUFFER_POOL, SAI_OBJECT_TYPE_INGRESS_PRIORITY_GROUP, SAI_OBJECT_TYPE_PORT, SAI_OBJECT_TYPE_QUEUE
+     */
+    SAI_TAM_COUNTER_SUBSCRIPTION_ATTR_OBJECT_ID,
+
+    /**
+     * @brief Subscribed stat enum
+     *
+     * @type sai_uint32_t
+     * @flags MANDATORY_ON_CREATE | CREATE_ONLY
+     */
+    SAI_TAM_COUNTER_SUBSCRIPTION_ATTR_STAT_ID,
```

Comment:
Any reason behind why this is not a list of STAT IDs and its expected to create one TAM subscription object per object ?

Reply:
Yes, to allow for a user label to be assigned per counter. Very early we thought to have a structure (possibly in a list) rather than an object per monitored counter. However, it is disallowed to reference object-ids in a structure. That is enforced by the SAI metadata library. It makes sense from an object and dependency-management perspective.

This is like the next-hop group member object as far as the object model goes.